dashboard/test/models/section_test.rb

```diff
@@ -66,4 +66,58 @@ class SectionTest < ActiveSupport::TestCase
 
     assert !Section.exists?(section.id)
   end
+
+  test 'add_student adds student to section' do
+    section = create :section
+    student = create :student
+
+    # even when the student is in another section from a different teacher
+    create(:section).add_student student
+
+    assert_creates(Follower) do
+      section.add_student student
+    end
+    assert section.students.exists?(student.id)
+    assert_equal 2, Follower.count(student_user_id: student.id)
+  end
+
+  test 'add_student in another section from the same teacher moves student' do
+    teacher = create :teacher
+    student = create :student
+    original_section = create :section, user: teacher
+    original_section.add_student student
+    new_section = create :section, user: teacher
+
+    # Initially, student is in original_section
+    assert original_section.students.exists?(student.id)
+    refute new_section.students.exists?(student.id)
+
+    assert_does_not_create(Follower) do
+      new_section.add_student student
+    end
+
+    # Verify student has been moved to new_section
+    refute original_section.students.exists?(student.id)
+    assert new_section.students.exists?(student.id)
+  end
+
+  test 'add_student with move_for_same_teacher: false does not move student' do
```

Comment:
True. The section has a single user (i.e. owner / teacher) and multiple student_users (i.e. followers / students). Soon, sections will be allowed to have multiple users (owners) as well. The terminology is pretty confusing.

Here I was trying to stick with the existing terms. The point of this test case is merely that `section.add_student` honors the `move_for_same_teacher` bool param if it's supplied. The section doesn't really care if the user type is student or teacher. This is never validated. Teachers can (and do intentionally for workshops) join sections as followers.

We could more clearly express the intent by making them both teachers and calling one the `organizer` and the other the `attendee` or `follower`? What do you think?